docs/JestObjectAPI.md

```diff
@@ -669,7 +669,7 @@ Creates a mock function similar to `jest.fn` but also tracks calls to `object[me
 
 :::note
 
-By default, `jest.spyOn` also calls the **spied** method. This is different behavior from most other test libraries. If you want to overwrite the original function, you can use `jest.spyOn(object, methodName).mockImplementation(() => customImplementation)` or `jest.replaceProperty(object, methodName, jest.fn(() => customImplementation));`
+By default, `jest.spyOn` also calls the **spied** method. This is different behavior from most other test libraries. If you want to overwrite the original function, you can use `jest.spyOn(object, methodName).mockImplementation(() => customImplementation)`.
```

Comment:
This is how it sounded before #13496

```suggestion
By default, `jest.spyOn` also calls the **spied** method. This is different behavior from most other test libraries. If you want to overwrite the original function, you can use `jest.spyOn(object, methodName).mockImplementation(() => customImplementation)` or `object[methodName] = jest.fn(() => customImplementation)`.
```

Reply:
This behavior still working on `"jest": "^29.6.2"`:
```js
object[methodName] = jest.fn(() => customImplementation)
```

Let me show how things are working in my example:
```js
class Math {
  total;

  constructor() {}

  sum(num1, num2) {
    return num1 + num2;
  }
}
```

example.test.js
```js
math["sum"] = (num1, num2) => num1 * num2; // works well
jest.spyOn(math, "sum").mockImplementation((num1, num2) => num1 * num2); // works well
jest.replaceProperty(math, "total", 20); // works well
```

```js
jest.replaceProperty(math, "sum", jest.fn((num1, num2) => num1 * num2)); // Cannot replace the `sum` property because it is a function. Use `jest.spyOn(object, 'sum')` instead.
```